kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/publisher/CloudPublisher.java

```diff
@@ -0,0 +1,96 @@
+package org.eclipse.kura.cloudconnection.publisher;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloudconnection.CloudConnectionManager;
+import org.eclipse.kura.cloudconnection.listener.CloudConnectionListener;
+import org.eclipse.kura.cloudconnection.listener.CloudDeliveryListener;
+import org.eclipse.kura.cloudconnection.message.KuraMessage;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * The {@link CloudPublisher} interface is an abstraction on top of the {@link CloudConnectionManager} to simplify the
+ * publishing process for each application running in the framework.
+ * CloudPublisher is used to publish a message with the specified KuraPayload onto a cloud connection.
+ * The {@link CloudPublisher} implementation abstracts to the applications all the low level specificities, like the
+ * message destination address, quality of service or properties because those are added by the {@link CloudPublisher}
+ * implementation based, for example, on a configuration.
+ *
+ * When an application wants to publish, it has to take a {@link CloudPublisher} instance and use the
+ * {@link CloudPublisher#publish(KuraMessage)} method, passing as argument a {@link KuraMessage}.
+ *
+ * Every {@link KuraMessage} accepted by the {@link CloudPublisher} is associated to a string identifier that can be
+ * used to confirm that the {@link KuraMessage} has been published. However, the semantics of the confirmation depends
+ * on both the implementation and the configuration of the publisher.
+ * For example, if the protocol of the cloud connection supports message identifiers and acknowledgments, the
+ * implementation will map the message identifier to the {@link KuraMessage} identifier and confirm it when
+ * the message identifier is acknowledged. If the protocol does not support message identifiers, or the message does not
+ * request an acknowledge, a confirmed {@link KuraMessage} identifier may at most indicate that the message has been
+ * successfully transmitted. There is no guarantee that a {@link KuraMessage} identifier will ever be confirmed. It is
+ * important that the implementation of the {@link CloudPublisher} and its configuration match the assumptions of the
+ * API consumer about the delivery.
+ *
+ * For example, if the correct behavior of an application requires guaranteed message delivery, the
+ * application should be configured to use a reliable publisher. Of course, applications that do not require this will
+ * work with any publisher.
+ *
+ * @since 2.0
+ * @noimplement This interface is not intended to be implemented by clients.
+ */
+@ProviderType
+public interface CloudPublisher {
+
+    /**
+     * Publishes the received {@link KuraMessage} using the associated Cloud Stack.
+     *
+     * @param message
+     *            The {@link KuraMessage} to be published
+     * @return a String representing the message ID
```

Comment:
`null` is returned if the Cloud Publisher will not confirm the message delivery, either because this is not supported by the underlying protocol or the Cloud Publisher itself is not implemented or configured to request the confirmation.